src/Bank/BankLogic/Bank.cs

```diff
@@ -3,40 +3,48 @@
 using System.Linq;
 using System.Text;
 
-namespace BankingSystem.BusinessLogic
+namespace BankingSystem.Bank
 {
     public class Bank
     {
-        private Dictionary<int, BankAccount> accounts;
+        private Dictionary<int, IBankAccount> accounts;
         private int nextAccount;
 
 
-        public Bank(Dictionary<int, BankAccount> accounts)
+        public Bank()
         {
-            this.accounts = accounts;
+            this.accounts = new Dictionary<int, IBankAccount>();
             this.nextAccount = 0; // Could be with Dependency injection
-            InterestRate = 0.01;
         }
 
-        public double InterestRate { get; }
+        private int NextAccountNumber => nextAccount++;
```

Comment:
Why do we need a property NextAccountNumber and a field NextAccount, I think it will be enough to have private get set property

Reply:
I think that maintain the a++ behavior is not very save each time I create a new account (How you see down), I thought that in the future I can create a new type of account or more than ones.
```
type switch
            {
                AccountType.CheckingAccount => new CheckingAccount(NextAccountNumber, origin),
                AccountType.SavingsAccount  => new SavingsAccount(NextAccountNumber, origin),
                _                           => throw new ArgumentException(message: "invalid enum value", paramName: nameof(type)),
                
            };
```

Reply:
Let me think about it